packages/buidler-etherscan/src/solc/bytecode.ts

```diff
@@ -0,0 +1,128 @@
+import { RunTaskFunction } from "@nomiclabs/buidler/types";
+import { InferralType } from "./SolcVersions";
+import {
+  CompilerInput,
+  CompilerOutput,
+  CompilerOutputBytecode
+} from "@nomiclabs/buidler/src/internal/buidler-evm/stack-traces/compiler-types";
+// At this point, this module was already imported.
+import { readSolcMetadataLength } from "./metadata";
+
+export class ContractNotFoundError extends Error {
+  constructor(message: string) {
+    super(message);
+  }
+}
+
+
+export async function lookupMatchingBytecode(contractFiles: CompilerOutput['contracts'], deployedBytecode: Buffer, inferralType: InferralType) {
+  for (const [ contractFilename, contracts ] of Object.entries(contractFiles)) {
+    for (const [ contractName, contract ] of Object.entries(contracts)) {
+      // Normalize deployed bytecode according to this contract.
+      const { deployedBytecode: runtimeBytecode } = contract.evm;
+
+      const runtimeBytecodeObject = Buffer.from(runtimeBytecode.object, "hex");
+      if (runtimeBytecodeObject.length != deployedBytecode.length) continue;
+
+      const {
+        immutableValues,
+        libraryLinks,
+        normalizedBytecode
+      } = await normalizeBytecode(deployedBytecode, runtimeBytecode);
```

Comment:
I must be missing something here, but how is it possible that `normalizeBytecode` is only called once?

My interpretation is that there are two things that need to be done: zeroing out the immutable references slices for the deployed bytecode, and adding the linked addresses to the runtime bytecode. That's why it seems weird to me that only one call is done (and one normalized bytecode is returned).